### ISSUE TITLE
[a11y]: cds-text-input labelling issues

### ISSUE THREAD

**Issue by shleewhite** [carbon-design-system/carbon#22332]
### Package

@carbon/web-components

### Package version

v2.55.0

### Automated testing tool and ruleset

Manual

### Assistive technology

JAWS, VoiceOver

### Description

Currently, the Carbon Web Component cds-text-input does not correctly associate the label and helper/error text with the input.

This is a critical error as important context is missing for screen reader users.

The `for` attribute is not set to the id of the input and the input is missing the `aria-describedby` attribute set to the id of the helper/error text.

Currently, the inputs are "labelled" because of the presence of placeholder text, so the automated tests can pass. This creates a couple difficulties:
* Assistive technology has spotty support for announcing placeholder text at all
* Usually, placeholder text is a sample of what the input content _should look like_, not the a description of what the inputs purpose is

Example: this is a screen shot from [the storybook](https://web-components.carbondesignsystem.com/?path=/docs/components-text-input--overview), in the accessibility tree the input is called "Placeholder text" not "Label text".

<img width="495" height="508" alt="Image" src="https://github.com/user-attachments/assets/cc25b606-c71c-4b8d-a7a9-d2724689ad88" />

### WCAG 2.1 Violation

4.1.2

### Reproduction/example

https://web-components.carbondesignsystem.com/?path=/docs/components-text-input--overview

### Steps to reproduce

1. Look at the public web component storybook page for text input
2. Inspect the label, notice that the for attribute is missing from the label element within the shadow dom
3. Inspect the input, notice the aria-describedby attribute is there, but not set to the id "helper-text", which is the id of the helper text

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
